Source/GRChomboCore/ChomboParameters.hpp

```diff
@@ -102,30 +75,100 @@ class ChomboParameters
                 }
             }
         }
-        pout() << "Center has been set to: ";
-        FOR1(idir) { pout() << center[idir] << " "; }
-        pout() << endl;
-
         if (nonperiodic_boundaries_exist)
         {
             // write out boundary conditions where non periodic - useful for
             // debug
             BoundaryConditions::write_boundary_conditions(boundary_params);
         }
 
+        // Grid N, L and center
+        int N_full;
+        pp.load("N_full", N_full, -1);
+
+        ivN = IntVect::Unit;
+        int max_N = N_full;
+        if (N_full != -1)
+        {
+            for (int dir = 0; dir < CH_SPACEDIM; ++dir)
+            {
+                if (boundary_params.lo_boundary[dir] ==
+                        BoundaryConditions::REFLECTIVE_BC ||
+                    boundary_params.hi_boundary[dir] ==
+                        BoundaryConditions::REFLECTIVE_BC)
+                {
+                    CH_assert(N_full % 2 == 0); // N_full is even
+                    ivN[dir] = N_full / 2 - 1;
+                }
+                else
+                    ivN[dir] = N_full - 1;
+            }
+            // halve max_N if fully reflective (reflective BC in all directions)
+            if (num_sym_bc == CH_SPACEDIM)
+                max_N /= 2;
+        }
```

Comment:
Can you make these compatible with non-cubic grids with symmetric BCs so that one can specify the `N_full` in each direction?